azurerm/resource_arm_mariadb_server.go

```diff
@@ -0,0 +1,381 @@
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(mariadb.Basic),
+								string(mariadb.GeneralPurpose),
+								string(mariadb.MemoryOptimized),
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
+
+						"family": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"administrator_login": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"administrator_login_password": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+
+			"storage_profile": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"storage_mb": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validate.IntBetweenAndDivisibleBy(5120, 4194304, 1024),
+						},
+
+						"backup_retention_days": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntBetween(7, 35),
+						},
+
+						"geo_redundant_backup": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(mariadb.Enabled),
+								string(mariadb.Disabled),
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
+					},
+				},
+			},
+
+			"ssl_enforcement": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(mariadb.SslEnforcementEnumDisabled),
+					string(mariadb.SslEnforcementEnumEnabled),
+				}, true),
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+
+			"fqdn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+
+		CustomizeDiff: func(diff *schema.ResourceDiff, v interface{}) error {
+
+			tier, _ := diff.GetOk("sku.0.tier")
+			storageMB, _ := diff.GetOk("storage_profile.0.storage_mb")
+
+			if strings.ToLower(tier.(string)) == "basic" && storageMB.(int) > 1048576 {
+				return fmt.Errorf("basic pricing tier only supports upto 1,048,576 MB (1TB) of storage")
+			}
+
+			return nil
+		},
+	}
+}
+
+func resourceArmMariaDbServerCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).mariadbServersClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM MariaDB Server creation.")
+
+	name := d.Get("name").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	adminLogin := d.Get("administrator_login").(string)
+	adminLoginPassword := d.Get("administrator_login_password").(string)
+	sslEnforcement := d.Get("ssl_enforcement").(string)
+	version := "10.2"
```

Comment:
IMO it'd be worth exposing this and making this a Required field, with validation just for `10.2` at this point. Whilst that's not ideal from a UX perspective - if we made that Optional+Defaulted to `10.2` when other versions come along we'd be defaulting to an old version)

Reply:
Added version back into schema as a required attribute, also updated the documentation to reflect this change.